utils.sh

```diff
@@ -380,18 +380,43 @@ EOF
 function verify_pull_secret() {
   # Do some PULL_SECRET sanity checking
   if [[ "${OPENSHIFT_RELEASE_IMAGE}" == *"registry.svc.ci.openshift.org"* ]]; then
-      if [[ "${PULL_SECRET}" != *"registry.svc.ci.openshift.org"* ]]; then
-          echo "Please get a valid pull secret for registry.svc.ci.openshift.org."
+      if [[ -z "${CI_TOKEN}"  ]]; then
+          error "Please login to https://api.ci.openshift.org and copy the token from the login command from the menu in the top right corner to set CI_TOKEN."
           exit 1
       fi
   fi
 
-  if [[ "${PULL_SECRET}" != *"cloud.openshift.com"* ]]; then
-      echo "Please get a valid pull secret for cloud.openshift.com."
+  if [[ "${PERSONAL_PULL_SECRET}" != *"cloud.openshift.com"* ]]; then
+      error "Get a valid pull secret (json string) from https://cloud.redhat.com/openshift/install/pull-secret"
       exit 1
   fi
 }
 
+function build_pull_secret() {
+    # Check to see if we already have a full pull secret.
+    # avoid "-z $PULL_SECRET" to ensure the secret is not logged
```

Comment:
Won't we also have a problem with CI_TOKEN and PERSONAL_PULL_SECRET being logged in `verify_pull_secret` above?

Reply:
We're logging PULL_SECRET every time we use it via echo anyway, so I don't know how accurate this comment really is.

Reply:
Another reason to have it in a file maybe.

Reply:
Yes, I think moving it to a file resolved that.